src/helpers/rgb.js

```diff
@@ -0,0 +1,48 @@
+// @flow
+
+/**
+ * Returns a string value for the color. The returned result is the smalles possible hex notation.
+ *
+ * @example
+ * // Styles as object usage
+ * const styles = {
+ *   background: rgb(255, 205, 100),
+ *   background: rgb({ red: 255, green: 205, blue: 100 }),
+ * }
+ *
+ * // styled-components usage
+ * const div = styled.div`
+ *   background: ${rgb(255, 205, 100)};
+ *   background: ${rgb({ red: 255, green: 205, blue: 100 })};
+ * `
+ *
+ * // CSS in JS Output
+ *
+ * element {
+ *   background: "#ffcd64";
+ *   background: "#ffcd64";
+ * }
+ */
+
+type RgbColor = {
+  red: number,
+  green: number,
+  blue: number,
+}
+
+function toHex(value) {
+  const hex = value.toString(16)
+  return hex.length === 1 ? `0${hex}` : hex
```

Comment:
If you create the hex of 0 it will be `'0'`. The issue here at hand is that only of 3 parts of a hex color can be reduced only then it should be. Otherwise two characters e.g. prefixed with a `0` are required.